### PR TITLE
Auto download latest Playback Dolphin and auto update both dolphins

### PR DESCRIPTION
**Pull request by NikhilNarayana**
~~No version checking logic yet (playback doesn't support it yet), should be added before release since 2.3.0 should be the norm and that will require a playback update~~

### CODE REVIEW

**Comment by NikhilNarayana** [comment]
added version checking cause it made sense to just do now, however this change kinda forces us to use node 12+ cause I needed spawnSync. Nothing seems to break, so it should be ok

File: src/renderer/lib/directories.ts [hunk 1]
```diff
@@ -15,12 +15,15 @@ export function getDefaultRootSlpPath(): string {
   return path.join(root, "Slippi");
 }
 
-export async function findDolphinExecutable(): Promise<string> {
+export async function findDolphinExecutable(
+  type: DolphinType
+): Promise<string> {
   // Make sure the directory actually exists
-  await fs.ensureDir(NETPLAY_PATH);
+  const dolphin_path = path.join(remote.app.getPath("userData"), type);
```
Comment (vinceau) [inline src/renderer/lib/directories.ts:22]: We should probably should stick with camelCase for variable names.

```suggestion
  const dolphinPath = path.join(remote.app.getPath("userData"), type);
```

File: src/renderer/lib/downloadDolphin.ts [hunk 2]
```diff
@@ -1,44 +1,84 @@
...
 ): Promise<void> {
   try {
-    await findDolphinExecutable();
-    log("Found existing Dolphin executable.");
+    await findDolphinExecutable(type);
+    log(`Found existing ${type} Dolphin executable.`);
+    log(`Checking if we need to update`);
+    const data = await getLatestReleaseData(type);
+    const latest_version = data.tag_name?.substring(1);
+    const is_outdated = await compareDolphinVersion(type, latest_version);
```
Comment (vinceau) [inline src/renderer/lib/downloadDolphin.ts:22]: You don't need to manually split out the `v` from the semver string. When using the `semver` package, it's smart enough to compare strings with or without the `v` prefix and the `compareDolphinVersion` calls `lt` from the `semver` package.

Also something something camelCase.

```suggestion
    const latestVersion = data.tag_name?.substring(1);
    const isOutdated = await compareDolphinVersion(type, latest_version);
```

File: src/renderer/lib/downloadDolphin.ts [hunk 3]
```diff
@@ -79,27 +120,38 @@ async function downloadLatestNetplay(
...
       const zip = new AdmZip(assetPath);
-      log(`Extracting to: ${extractToLocation}, and renaming to netplay`);
-      zip.extractAllTo(extractToLocation, true);
-      const oldPath = path.join(extractToLocation, "FM-Slippi");
-      const newPath = NETPLAY_PATH;
-      if (await fs.pathExists(newPath)) {
-        log(`${newPath} already exists. Deleting...`);
-        await fs.remove(newPath);
+      const already_installed = await fs.pathExists(dolphin_path);
+      if (already_installed) {
```
Comment (vinceau) [inline src/renderer/lib/downloadDolphin.ts:139]: 🐫

```suggestion
      const alreadyInstalled = await fs.pathExists(dolphin_path);
      if (alreadyInstalled) {
```

**Review by vinceau — CHANGES_REQUESTED** [review]
Only some minor changes.

File: src/renderer/lib/downloadDolphin.ts (hunk 2 above)
Comment (NikhilNarayana) [reply src/renderer/lib/downloadDolphin.ts:22]: woah, that's nice

**Review by vinceau — APPROVED** [review]
LGTM!

**Review by vinceau — APPROVED** [review]

**Review by vinceau — APPROVED** [review]

File: src/renderer/lib/downloadDolphin.ts [hunk 4]
```diff
@@ -126,70 +126,43 @@ async function installDolphin(
...
-        await fs.move(
-          dolphinPath,
-          path.join(remote.app.getPath("userData"), type + "_old")
-        );
-      }
+      const backupLocation = path.join(
+        remote.app.getPath("userData"),
+        type + "_old"
+      );
+
```
Comment (vinceau) [inline src/renderer/lib/downloadDolphin.ts:139]: Probably don't need to declare this twice.

**Review by vinceau — CHANGES_REQUESTED** [review]

**Review by vinceau — APPROVED** [review]

**Pull request merged**